### PR TITLE
Changed phase status filtering to work with init containers as well

### PR DESCRIPTION
**Pull request by yarinm**
Fix for https://github.com/Azure/aad-pod-identity/issues/169

Note that you can't use OR in selector interface but AND with NOT does work so i created an ignore list for statuses with don't want pod-identity to work on.

### CODE REVIEW

File: pkg/k8s/client.go [hunk 1]
```diff
@@ -21,6 +22,9 @@ const (
 	getPodListSleepTimeMilliseconds = 300
 )
 
+// We only want to allow pod-identity with Pending or Running phase status
+var IGNORED_POD_PHASE_STATUSES = []string{"Succeeded", "Failed", "Unknown", "Completed", "CrashLoopBackOff"}
```
Comment (kkmsft) [inline pkg/k8s/client.go]: Can you please move this to different casing ? 
May be: var ignoredPodPhases = []string {...}

File: pkg/k8s/client.go [hunk 2]
```diff
@@ -76,8 +80,9 @@ func (c *KubeClient) GetPodName(podip string) (podns, poddname string, err error
 }
 
 func (c *KubeClient) getPodListWithTries(podip string, tries int) (*v1.PodList, error) {
+	phaseFilter := ",status.phase!=" + strings.Join(IGNORED_POD_PHASE_STATUSES, ",status.phase!=")
```
Comment (kkmsft) [inline pkg/k8s/client.go]: Is there a way we can do this without recomputing the filter every time ?

File: pkg/k8s/client.go [hunk 3]
```diff
@@ -21,6 +22,9 @@ const (
 	getPodListSleepTimeMilliseconds = 300
 )
```
Comment (kkmsft) [inline pkg/k8s/client.go]: First of all - Thank you @yarinm for raising the issue and this PR.

File: pkg/k8s/client.go (hunk 2 above)
Comment (yarinm) [reply pkg/k8s/client.go]: I’m no golang expert but I think there are two possible solutions here:
1. Write the filter string in a hard-coded way
2. Create a package wide variable For the filter and create a getOrCreateFilter method that will initialize the variable once and return its value

WDYT?

File: pkg/k8s/client.go (hunk 3 above)
Comment (yarinm) [reply pkg/k8s/client.go]: @kkmsft I went over your comments and resolved the issues, let me know what you think.

**Comment by yarinm** [comment]
@kkmsft ? any updates about this? can we merge it?

**Comment by kkmsft** [comment]
@yarinm - Apologies for the delay. Initially I was hoping to get the e2e stabilized and then merge this. But that is going to take some time. Right now there is azure pipeline failure with this. I will fix that first and then get it merged this week. Thank you very much for this work.

**Comment by kkmsft** [comment]
@yarinm - can you please rebase your branch to the latest ?

**Comment by yarinm** [comment]
I'm closing this PR, merge https://github.com/Azure/aad-pod-identity/pull/191

**Pull request closed without merging**